custom_components/solarman/sensor.py

```diff
@@ -186,6 +186,8 @@ def update(self):
             if self._field_name in val:
                 self.p_state = val[self._field_name]
             else:
+                if self.uom in ("W", "kW", "°C", "Hz", "V", "A", "VA", "%"):
```

Comment:
Parameters with `isstr: true` set (e.g. parameter `Hardware Version` in `deye_2mppt.yaml`) will throw an AttributeError here when their states get invalidated. The attribute `self.uom` doesn't exist for these.

Example AttributeError:
```
2023-05-16 23:35:05.931 ERROR (MainThread) [homeassistant.helpers.entity] Update for sensor.pv_hardware_version fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 554, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 781, in async_device_update
    await hass.async_add_executor_job(self.update)
  File "/usr/local/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/solarman/sensor.py", line 189, in update
    if self.uom in ("W", "kW", "°C", "Hz", "V", "A", "VA", "%"):
AttributeError: 'SolarmanSensorText' object has no attribute 'uom'
```
As a quick fix, one could check for attribute existence here:
```suggestion
                if hasattr(self, "uom") and self.uom in ("W", "kW", "°C", "Hz", "V", "A", "VA", "%"):
```